src/Symfony/Component/Console/Application.php

```diff
@@ -526,20 +526,25 @@ public function find($name)
         // filter out aliases for commands which are already on the list
         if (count($commands) > 1) {
             $commandList = $this->commands;
-            $commands = array_filter($commands, function ($nameOrAlias) use ($commandList, $commands) {
+            $commands = array_filter($commands, function ($nameOrAlias) use ($commandList, $commands, &$aliases) {
                 $commandName = $commandList[$nameOrAlias]->getName();
+                $aliases[$nameOrAlias] = $commandName;
 
                 return $commandName === $nameOrAlias || !in_array($commandName, $commands);
             });
         }
 
         $exact = in_array($name, $commands, true);
-        if (count($commands) > 1 && !$exact) {
+        if (!isset($aliases[$name]) && count($commands) > 1 && !$exact) {
```

Comment:
changing line 537 to `$exact = \in_array($name, $commands, true) || isset($aliases[$name]);` would avoid the need for both this check and the condition added below

Reply:
it's betterl ike this ;).